src/Modules/OrchardCore.Commerce.Payment/Settings/CheckoutAddressSettingsDisplayDriver.cs

```diff
@@ -0,0 +1,66 @@
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Http;
+using OrchardCore.DisplayManagement.Entities;
+using OrchardCore.DisplayManagement.Handlers;
+using OrchardCore.DisplayManagement.Views;
+using OrchardCore.Environment.Shell;
+using OrchardCore.Settings;
+using System.Threading.Tasks;
+
+namespace OrchardCore.Commerce.Payment.Settings;
+
+public class CheckoutAddressSettingsDisplayDriver : SiteDisplayDriver<CheckoutAddressSettings>
+{
+    public const string GroupId = "CheckoutAddressSettings";
+
+    private readonly IShellHost _shellHost;
+    private readonly ShellSettings _shellSettings;
+    private readonly IHttpContextAccessor _hca;
+    private readonly IAuthorizationService _authorizationService;
+
+    public CheckoutAddressSettingsDisplayDriver(
+        IShellHost shellHost,
+        ShellSettings shellSettings,
+        IHttpContextAccessor hca,
+        IAuthorizationService authorizationService)
+    {
+        _shellHost = shellHost;
+        _shellSettings = shellSettings;
+        _hca = hca;
+        _authorizationService = authorizationService;
+    }
+
+    protected override string SettingsGroupId => GroupId;
+
+    public override async Task<IDisplayResult> EditAsync(ISite model, CheckoutAddressSettings section, BuildEditorContext context)
+    {
+        if (!await AuthorizeAsync())
+        {
+#pragma warning disable CS8603
+            return null;
+#pragma warning restore CS8603
+        }
```

Comment:
Always include a comment explaining why you use pragma warning disable.
But in this case, I think you can just do this. (note the added `?` in the `Task<IDisplayResult?>` return type)

```csharp
    public override async Task<IDisplayResult?> EditAsync(ISite model, CheckoutAddressSettings section, BuildEditorContext context)
    {
        if (!await AuthorizeAsync()) return null;
```